src/OpenCL.cpp

```diff
@@ -412,26 +412,24 @@ const std::string sourceCode_sgemm =
 ;
 #endif
 
-thread_local ThreadData opencl_thread_data;
-
-void OpenCL::ensure_thread_initialized() {
-    if (!opencl_thread_data.m_is_initialized) {
+void OpenCL::ensure_thread_initialized(OpenCLContext &opencl_context) {
```

Comment:
If this takes a context argument it is no longer thread specific, so the function name is wrong.

Reply:
renaming to ensure_thread_initialized